cmd/server/main.go

```
@@ -1,29 +1,115 @@
 package main
 
 import (
+	"fmt"
 	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
 
 	"github.com/go-chi/chi/v5"
 
 	"github.com/zhenyanesterkova/metricsmonitor/internal/app/server/config"
-	"github.com/zhenyanesterkova/metricsmonitor/internal/handlers"
+	"github.com/zhenyanesterkova/metricsmonitor/internal/app/server/logger"
+	"github.com/zhenyanesterkova/metricsmonitor/internal/app/server/rwfile"
+	"github.com/zhenyanesterkova/metricsmonitor/internal/handler"
 	"github.com/zhenyanesterkova/metricsmonitor/internal/storage/memstorage"
 )
 
 func main() {
+	if err := run(); err != nil {
+		fmt.Printf("%v", err)
+		os.Exit(1)
+	}
+}
+
+func run() error {
+	cfg := config.New()
+	err := cfg.Build()
+	if err != nil {
+		fmt.Printf("can not build config: %v", err)
+		return fmt.Errorf("config error: %w", err)
+	}
 
-	cfgBuilder := config.GetConfigBuilder()
-	cfg := cfgBuilder.Build()
+	loggerInst := logger.NewLogrusLogger()
+	err = loggerInst.SetLevelForLog(cfg.LConfig.Level)
+	if err != nil {
+		loggerInst.LogrusLog.Errorf("can not parse log level: %v", err)
```

Здесь лучше вернуть ошибку и завершить работу приложения: если при иницализации сервиса возникла ошибка, то лучше fail early

---

Сделано